### ISSUE TITLE
Windows: Scrolling is not smooth but lags

### ISSUE THREAD

**Issue by gurpreetshanky** [Microsoft/vscode#61824]
- VSCode Version: Vscode Insiders ,Vscode Stable
- OS Version: Windows 10 (Fast Ring)

Steps to Reproduce:

1. Launch vscode
2. If minimize, scrolling shutters after maximize

Does this issue occur when all extensions are disabled?: Yes

This is a duplicate of #13612 but I couldn't post it there as it was closed.

**Comment by gurpreetshanky** [comment]
Version 1809 ( OS Build 18267.1001)

**Comment by bpasero** [comment]
@gurpreetshanky interesting, that should be Windows 10 RS5. Can you give more details about:
* what device are you using
* how do you scroll (external mouse or touchpad)
* does it reproduce if you use Code (stable) and configure the `smoothScrollingWorkaround` to true

**Comment by gurpreetshanky** [comment]
This is the Dell XPS 15 with Microsoft Precision touchpad.
Yes it works with smoothscrollingworkaround.

This is the same bug which I have linked #13612. Most probably, it has returned in insider version of Windows.

**Comment by bpasero** [comment]
@gurpreetshanky that would be bad, any chance you could test the non-insiders version?

**Comment by gurpreetshanky** [comment]
Yes this problem is not present in stable version. Has smoothScrollingWorkaround setting is removed in insiders version? I cannot find it.

**Comment by bpasero** [comment]
@gurpreetshanky yes I removed it assuming that this issue was fixed in Windows 10 RS5. It is unlikely I would bring it back, we should rather find out why this is broken again in Windows insiders. I wonder if your insiders version is somehow behind the RS5 update.

**Comment by gurpreetshanky** [comment]
@bpasero No mine is fast ring(18267), so its after the RS5 update. This bug definitely came back in new Windows Insiders version.

**Comment by bpasero** [comment]
I will follow up with the developer that pushed the fix to ask if this possibly had to be reverted for some reason

**Comment by gurpreetshanky** [comment]
@rossnichols here is the file https://1drv.ms/u/s!AgmFl0OpBO1agYtxQUmHZHEtPL8dlg

**Comment by rossnichols** [comment]
@gurpreetshanky thanks! Root cause is known, I'll work with Ben on getting VSCode fixed.

**Comment by korpong753** [comment]
So, What is the conclusion after you have know root cause. Do I need also to follow your step below to solving problem or I have to wait for your fixed VSCode and release its in next patch.

![image](https://user-images.githubusercontent.com/42170550/47613091-bd547080-daba-11e8-844e-889e1d84acbd.png)

**Comment by rossnichols** [comment]
@korpong753, you don't need to follow the instructions, I have all the information I needed. You'll just have to wait until VSCode has an update that fixes it.

**Comment by korpong753** [comment]
@rossnichols, I'm not sure, Is it bug of VSCode or not, Once we have checked and updated window 10 pro, now we can use touchpad with smoothly without lag problem anymore.

This is window version that we updated. (https://support.microsoft.com/th-th/help/4462933/windows-10-update-kb4462933)

![image](https://user-images.githubusercontent.com/42170550/47629557-5c867000-db6e-11e8-895d-b032291fc0ec.png)

**Comment by rossnichols** [comment]
@korpong753, can you collect a trace using the instructions above?

**Comment by HazemAM** [comment]
@rossnichols I can confirm scrolling is much better with a precision touchpad after the KB4462933 update with minimal lagging (still lags a little bit, but it's not affected by VSCode's focus change, AFAICT).

Windows & VSCode both on stable.

**Comment by bpasero** [comment]
@gurpreetshanky can I ask you if the scroll-lag workaround setting in VSCode fixes the issue for your? Just configure:
* `window.smoothScrollingWorkaround: true`
* `window.titleBarStyle: native`

**Comment by gurpreetshanky** [comment]
@bpasero Yes it has fixed the issue for me by these workarounds.

**Comment by bpasero** [comment]
@gurpreetshanky @korpong753 @HazemAM can you help me to test a new build with the fix from https://chromium-review.googlesource.com/c/chromium/src/+/1299342 to see if it works now without having `window.smoothScrollingWorkaround: true` configured?

**Steps:**
* [download](https://az764295.vo.msecnd.net/insider/5967828c2a5c559ed86c3369ed9811bb1bd6386b/VSCode-win32-x64-1.31.0-insider.zip)
* open settings and configure `window.smoothScrollingWorkaround: false`
* restart Code
* see if scrolling is smooth

**Comment by gurpreetshanky** [comment]
@bpasero Honestly I tried both the Insiders versions (normal and above provided one) and checked multiple times. The workaround was not required in both situations.

**Comment by bpasero** [comment]
@gurpreetshanky great, thanks for confirming. By now the current insiders version and the one I linked will both have the fix included :+1:
